cmd/migrate/list_item_example_test.go

```diff
@@ -0,0 +1,88 @@
+package main
+
+import "testing"
+
+func TestListItemTransformation(t *testing.T) {
+	tests := []TestCase{
+		{
+			Name: "",
+			Config: `
+resource "cloudflare_list" "known_openai_ips" {
+  account_id  = local.account_id
+  name        = "known_openai_ips"
+  description = "The set of IPs that internal OpenAI requests may come from. Used for exemption from firewall and bot detection rules. Do not use for general access control!"
+  kind        = "ip"
+
+  dynamic "item" {
+    for_each = local.ips_data.ips
+    iterator = ip_entry
+    content {
+      comment = ip_entry.value.comment
+
+      value {
+        ip = ip_entry.value.ip
+      }
+    }
+  }
+}
+`,
+			Expected: []string{`
+resource "cloudflare_list" "known_openai_ips" {
+  account_id  = local.account_id
+  description = "The set of IPs that internal OpenAI requests may come from. Used for exemption from firewall and bot detection rules. Do not use for general access control!"
+  kind        = "ip"
+  name        = "known_openai_ips"
+}
+`, `
+resource "cloudflare_list_item" "known_openai_ips" {
+  account_id = cloudflare_list_item.known_openai_ips.account_id
+  comment    = each.value.content.comment
+  for_each   = [for ip_entry in [for value in local.ips_data.ips : { key = value, value = value }] : { content = { comment = ip_entry.value.comment, value = { ip = ip_entry.value.ip } } }]
```

Comment:
the double for comprehension translation is ugly, but it handles almost arbitrary cases, since it does not require too much AST rewrite of the dynamic block's bodies